### PR TITLE
Bump docker images to `bullseye`

### PR DESCRIPTION
**Pull request by frouioui**
## Description

This Pull Request bumps the bootstrap version to `20` and changes the default Debian version used from `buster` to `bullseye` for all images.

## Checklist

-   [x] "Backport to:" labels have been added if this change should be back-ported
-   [x] Tests were added or are not required
-   [x] Did the new or modified tests pass consistently locally and on the CI
-   [x] Documentation was added or is not required

### CODE REVIEW

**Comment by dbussink** [comment]
> MySQL57 is not natively available in Debian Bullseye and Bookworm. All images using MySQL57, like `vitess/lite:mysql57`, will remain on the Buster version and will keep using `go1.20.5` for the entire lifecycle of `v18.0`.

I think we should install the Buster packages here. I don't think we should lock ourselves into Go 1.20 for the v18 release now already. 

See for example also here where we use an older distro version for CI:

https://github.com/vitessio/vitess/blob/54f0b33a652f929ecc43a949371d49364a3b94c8/.github/workflows/unit_test_mysql57.yml#L99-L107

**Comment by frouioui** [comment]
@dbussink I have changed the `mysql57` image to be built on `bullseye` but to download `buster` MySQL57 packages.

File: docker/bootstrap/CHANGELOG.md [hunk 1]
```diff
@@ -72,4 +72,8 @@ List of changes between bootstrap image versions.
 
 ## [19] - 2023-06-07
 ### Changes
-- Update build to golang 1.20.5
\ No newline at end of file
+- Update build to golang 1.20.5
+
+## [20] - 2023-08-03
+### Changes
+- Bump all images (besides mysql57) to bullseye base image
```
Comment (dbussink) [inline docker/bootstrap/CHANGELOG.md:79]: I think this is still stale?

**Review by dbussink — COMMENTED** [review]
Looks like we still have CI failures here, is that due to the bootstrap missing?

File: docker/bootstrap/CHANGELOG.md (hunk 1 above)
Comment (frouioui) [reply docker/bootstrap/CHANGELOG.md]: fixed via b6ce170

**Comment by frouioui** [comment]
> Looks like we still have CI failures here, is that due to the bootstrap missing?

Correct, CI ran before I was done pushing the images.

**Review by dbussink — APPROVED** [review]

**Review by systay — APPROVED** [review]

**Review by harshit-gangal — APPROVED** [review]

**Comment by harshit-gangal** [comment]
should be add more details in release note about this change?

**Comment by frouioui** [comment]
@harshit-gangal, there is already a note regarding the deprecation of `buster`:
```
#### <a id="debian-buster"/>Buster removed

Buster LTS supports will stop in June 2024, and Vitess v18.0 will be supported through October 2024. 
To prevent supporting a deprecated buster build for several months after June 2024, we are preemptively
removing Vitess support.
```

**Pull request merged**